src/uart.rs

```diff
@@ -752,30 +766,53 @@ impl<'d> UartDriver<'d> {
     /// Unlike [`split`], the halves are owned and reference counted.
     pub fn into_split(self) -> (UartTxDriver<'d>, UartRxDriver<'d>) {
         let port = self.port;
+        let tx_queue = self
+            .queue
+            .as_ref()
+            .map(|queue| unsafe { Queue::new_borrowed(queue.as_raw()) });
+        let rx_queue = self
+            .queue
+            .as_ref()
+            .map(|queue| unsafe { Queue::new_borrowed(queue.as_raw()) });
         let _ = ManuallyDrop::new(self);
         REFS[port as usize].fetch_add(2, Ordering::SeqCst);
         (
             UartTxDriver {
                 port,
                 owner: Owner::Shared,
+                queue: tx_queue,
                 _p: PhantomData,
             },
             UartRxDriver {
                 port,
                 owner: Owner::Shared,
+                queue: rx_queue,
                 _p: PhantomData,
             },
         )
     }
 
     /// Read multiple bytes into a slice
-    pub fn read(&self, buf: &mut [u8], delay: TickType_t) -> Result<usize, EspError> {
-        self.rx().read(buf, delay)
+    pub fn read(&self, buf: &mut [u8], timeout: TickType_t) -> Result<usize, EspError> {
```

Comment:
```suggestion
    pub fn read(&self, buf: &mut [u8], timeout: T) -> Result<usize, EspError> 
    where T: Into<TickType_t>
    {
```

What do you think about this?

Reply:
Sure - that would improve the ergonomics of the API. It is just that we then need to apply it everywhere where `TockType_t` is currently used. Perhaps a separate PR?